File: python/scrabble-score/scrabble_score.py
scores = { c: i for chars, i in [
    ("AEIOULNRST", 1),
    ("DG", 2),
    ("BCMP", 3),
    ("FHVWY", 4),
    ("K", 5),
    ("JX", 8),
    ("QZ", 10)
] for c in chars }

def score(chars):
    return sum([scores[c] for c in chars.upper()])

File: javascript/scrabble-score/scrabble-score.js
const scores = {
    1: 'AEIOULNRST',
    2: 'DG',
    3: 'BCMP',
    4: 'FHVWY',
    5: 'K',
    8: 'JX',
    10: 'QZ'
};

const scoreMap = Object.entries(scores)
    .reduce((acc, [score, [...letters]]) => {
        letters.forEach(l => acc[l] = score);
        return acc;
    }, {});

module.exports = (word) => {
    return (word || '')
        .toUpperCase()
        .split('')
        .reduce((acc, c) => {
            return acc + Number(scoreMap[c]);
        }, 0);
}